tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -2705,6 +2705,39 @@ write_verbose_type(LL_Type *ll_type)
   print_token(ll_type->str);
 }
 
+/* whether debug location should be suppressed */
+static bool should_suppress_debug_loc(INSTR_LIST *instrs) {
+  if (!instrs)
+    return false;
+
+  // return true if not a call instruction
+  switch (instrs->i_name) {
+  case I_INVOKE:
+    return false;
+  case I_CALL:
+    // f90 runtime functions fort_init and f90_* dont need debug location
+    if (instrs->prev && (instrs->operands->ot_type == OT_TMP) &&
+        (instrs->operands->tmps == instrs->prev->tmps) &&
+        (instrs->prev->operands->ot_type == OT_VAR)) {
+      // We dont need to expose those internals in prolog to user
+      // %1 = bitcast void (...)* @fort_init to void (i8*, ...)*
+      // call void (i8*, ...) %1(i8* %0)
+      // %8 = bitcast void (...)* @f90_template1_i8 to void (i8*, i8*, i8*, i8*,
+      //      i8*, i8*, ...)*
+      // call void (i8*, i8*, i8*, i8*, i8*, i8*, ...) %8(i8*
+      //      %2, i8* %3, i8* %4, i8* %5, i8* %6, i8* %7)
+
+      if (char *name_str = instrs->prev->operands->string) {
+        return (!strncmp(name_str, "@fort_init", strlen("@fort_init")) ||
+                !strncmp(name_str, "@f90_", strlen("@f90_")));
```

Comment:
What about fortran statements like allocate which in turn generate runtime calls, @f90_alloc*. Will this patch disable generation of line-numbers for these runtime calls also?

Reply:
Generation of debug info is suppressed only for the statements in prologue, other occurrences would have debug location.

Reply:
> 
> 
> What about fortran statements like allocate which in turn generate runtime calls, @f90_alloc*. Will this patch disable generation of line-numbers for these runtime calls also?

Generation of debug info is suppressed only for the statements in prologue, other occurrences would keep having debug location.